### PR TITLE
Added welcomemessages, ipinfo_whois.c, listsg.c.

### PR DESCRIPTION
**Pull request by revrsefr**
New module welcomemessages.c
New module listsg.c
New module ipinfo_whois.c

### CODE REVIEW

**Comment by syzop** [comment]
Heya, it's good to see another person starting with writing unrealircd modules.

A few points though (which aren't really about the modules themselves like what they do):

1. We had a talk about AI-generated code. It is my position that it is OK for third party modules. This is unrealircd-contrib, third party modules, in which case our job isn't really to go through the code and check that it all works perfectly and we don't really care about code style either. We mostly check that it isn't any evil code like one that rm -rf /'s :D. What still is good to mention is that, as the author you will have to fix serious bugs if people report them and we have this clause in the rules that if like serious crash bugs are reported and go unfixed for a longer period then we may remove the module. That's the only caveat I can think of: if you use AI to generate code, sure, but the module is still your responsibility and you should understand that if something bad happens like crash bugs you will be asked to fix things and should be able to. https://www.unrealircd.org/docs/Rules_for_3rd_party_modules_in_unrealircd-contrib
2. In the future, please file one PR per new module. As combining multiple files into a single PR makes it harder to reject/accept and deal with changes. I see this is nowhere stated in our rules or otherwise so I guess I will have to add that :)
3. It seems the files use the old m_ style prefix for modules that UnrealIRCd 3.2 had 4.x had. We dropped that with UnrealIRCd 5 onwards. As you can see none of the modules in https://modules.unrealircd.org/ have that anymore. Could you rename them to drop that m_ prefix (both in the file and the MOD_HEADER and comments) ?
4. It seems you sortof already did that with ipident.c and m_ipdent.c which occur twice in this PR? Can you remove the m_ipident.c one? (Assuming that is the old one and it is not the other way around)

**Comment by revrsefr** [comment]
Now should be ok.. :D

File: files/ipinfo_whois.c [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+*/
+/*** <<<MODULE MANAGER START>>>
+module
+{
+    documentation "https://github.com/revrsedev/UnrealIRCD-6-mods-contrib/blob/main/ipinfo_whois/README.md";
+    troubleshooting "In case of problems, documentation or e-mail me at mike.chevronnet@gmail.com";
+    min-unrealircd-version "6.*";
+    post-install-text {
+        "The module is installed, now all you need to do is add a 'loadmodule' line to your config file:";
+        "loadmodule \"third/m_ipinfo_whois\";";
```
Comment (syzop) [inline files/ipinfo_whois.c]: This still tells to loadmodule the wrong module name.

File: files/listsg.c [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+*/
+/*** <<<MODULE MANAGER START>>>
+module
+{
+    documentation "https://github.com/revrsedev/UnrealIRCD-6-mods-contrib/blob/main/listsg/README.md";
+    troubleshooting "In case of problems, documentation or e-mail me at mike.chevronnet@gmail.com";
+    min-unrealircd-version "6.*";
+    post-install-text {
+        "The module is installed, now all you need to do is add a 'loadmodule' line to your config file:";
+        "loadmodule \"third/m_listsg\";";
```
Comment (syzop) [inline files/listsg.c]: Here too, it tells to loadmodule the wrong module name.

**Comment by syzop** [comment]
The modules are in. For the ipinfo_whois if someone uses `REHASH` I think it will loose information and due to the direct use of CURL will crash if a request (https call) is in progress. Probably good idea to make the module permanent for now.

All that can be fixed with use of SavePersistentPointer() and stuff for the hash table and using https://www.unrealircd.org/docs/Dev:URL_API instead of the CURL API directly. But.. that is kinda more advanced.

Just adding a quick
```c
ModuleSetOptions(modinfo->handle, MOD_OPT_PERM, 1);
```
in the MOD_INIT() is probably a better idea for now.

Does mean they cannot reload the module though, but, like i said, otherwise people will crash, or risk to crash.

**Pull request merged**